File: classic puzzle-medium/ascii-art-glass-stacking.py
#https://www.codingame.com/training/medium/ascii-art-glass-stacking
#ASCII ART : GLASS STACKING
#concept: Ascii Art
import sys
import math

n=int(input())
eq=lambda a : int(eval(f"({a})*({a+1})*0.5"))
l=0
while eq(l)<n: l+=1
if eq(l)>n: l-=1
h=[" *** "," * * "," * * ","*****"]
for layer in range(1,l+1):
    spaces=3*(l-layer)*" "
    for i in range(0,4):
        s=" ".join([h[i]]*layer)
        print(spaces+s+spaces)

File: classic puzzle-medium/ascii-art-glass-stacking.rb
#https://www.codingame.com/training/medium/ascii-art-glass-stacking
#ASCII ART : GLASS STACKING
#concept: Ascii Art
n=gets.to_i
l=0
#define number of layers
eq=lambda{|l|eval("(#{l})*(#{l+1})*0.5").to_i}
l+=1 while eq.call(l)<n
l-=1 if eq.call(l)>n
#layers
h=[" *** "," * * "," * * ","*****"]
(1..l).each{|layer|
    spaces=3*(l-layer)
    (0...4).each{|i|puts " "*spaces+[h[i]]*layer*" "+" "*spaces}
}
